admin_tests/preprints/test_views.py

```diff
@@ -103,8 +103,8 @@ def test_get_object(self, req, preprint, plain_view):
     def test_no_user_permissions_raises_error(self, user, preprint, plain_view):
         request = RequestFactory().get(reverse('preprints:preprint', kwargs={'guid': preprint._id}))
         request.user = user
-        resp = plain_view.as_view()(request, guid=preprint._id)
```

Comment:
Tests failed with:
```
admin_tests/preprints/test_views.py:106: in test_no_user_permissions_raises_error
    resp = plain_view.as_view()(request, guid=preprint._id)
/usr/lib/python3.6/site-packages/django/views/generic/base.py:70: in view
    return self.dispatch(request, *args, **kwargs)
/usr/lib/python3.6/site-packages/django/contrib/auth/mixins.py:103: in dispatch
    return self.handle_no_permission()
/usr/lib/python3.6/site-packages/django/contrib/auth/mixins.py:46: in handle_no_permission
    raise PermissionDenied(self.get_permission_denied_message())
E   django.core.exceptions.PermissionDenied
```
Due to DRF changes

Reply:
There is a difference: the old test redirects to login while the new one raises exception? (though I feel the new one is the correct behavior but please double check.). We can also test this one on staging3.

Reply:
Yes, to be clear the permission error prevents the test from completing, so it doesn't get to the redirect. I think it should redirect normally.